src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerFunctionalTest.java

```diff
@@ -216,34 +216,53 @@ public void should_succeed_and_create_a_task_with_no_categories() {
             "Asylum"
         );
 
-        Response taskFound = findTasksByCaseId(caseId, 1);
-
-        caseId1Task1Id = taskFound
-            .then().assertThat()
-            .body("[0].id", notNullValue())
-            .extract()
-            .path("[0].id");
-
-        taskIdStatusMap.put(caseId1Task1Id, "completed");
+        await().ignoreException(AssertionError.class)
+            .pollInterval(2, SECONDS)
+            .atMost(180, SECONDS)
+            .until(
+                () -> {
+                    Response taskFound = findTasksByCaseId(caseId, 1);
+                    if (taskFound != null) {
+                        caseId1Task1Id = taskFound
+                            .then().assertThat()
+                            .body("[0].id", notNullValue())
+                            .extract()
+                            .path("[0].id");
 
-        Response response = findTaskDetailsForGivenTaskId(caseId1Task1Id);
-
-        response.then().assertThat()
-            .statusCode(HttpStatus.OK.value())
-            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
-            .body("caseTypeId.value", equalToIgnoringCase("asylum"))
-            .body("jurisdiction.value", equalToIgnoringCase("ia"))
-            .body("dueDate.value", notNullValue())
-            .body("taskState.value", equalToIgnoringCase("unconfigured"))
-            .body("hasWarnings.value", is(false))
-            .body("caseId.value", is(caseId))
-            .body("name.value", equalToIgnoringCase("Follow-up extended direction"))
-            .body("workingDaysAllowed.value", is(2))
-            .body("isDuplicate.value", is(false))
-            .body("delayUntil.value", notNullValue())
-            .body("taskId.value", equalToIgnoringCase("followUpExtendedDirection"))
-            .body("warningList.value", is("[]"));
+                        taskIdStatusMap.put(caseId1Task1Id, "completed");
```

Comment:
how do you know that the status is `completed` here?  Looks like you're only finding the `id`.